sdk/tvOS/OoyalaTVSkinSDK/Widgets/OOOoyalaTVClosedCaptionsView.m

```diff
@@ -0,0 +1,333 @@
+    self.backgroundView.hidden = YES;
+    self.backgroundView.layer.cornerRadius = 10;
+    self.backgroundView.layer.masksToBounds = YES;
+    
+    // setup UITextView according to style
+    self.textView = [[OOClosedCaptionsTextView alloc] initWithFrame:frame style:style backgroundView:self.backgroundView];
+    self.textView.autoresizingMask = UIViewAutoresizingFlexibleTopMargin|UIViewAutoresizingFlexibleLeftMargin|UIViewAutoresizingFlexibleRightMargin;
+    
+    [self addSubview:self.backgroundView];
+    [self addSubview:self.textView];
+    
+    [self setNeedsDisplay];
+}
+
+- (void)drawRect:(CGRect)rect {
+    //do not draw if no text
+    if (self.caption.text == nil) {
+        self.textView.hidden = YES;
+        self.backgroundView.hidden = YES;
+        return;
+    }
+    
+    if (self.textView.hidden || self.backgroundView.hidden) {
+        self.textView.hidden = NO;
+        self.backgroundView.hidden = NO;
+    }
+    
+    [self matchFrameWithText];
+    
+    // update highlight
+    self.backgroundView.textRects = [self.textView getRectsForEachLine:[self.splitText componentsSeparatedByString:@"\n"]];
+    [self.backgroundView updateBackground];
+}
+
+// Set UITextView frame based on current closed caption text size
+// This is for both pop-on and paint-on
+- (void)matchFrameWithText {
+    
+    // Set maxWidth based on the self.frame (the passed frame in init function is the frame of player)
+    CGFloat maxWidth = 0.0;
+    maxWidth = self.frame.size.width * 0.9;
+    
+    
+    [self.textView setFont:self.style.textFontName frame:self.frame  baseFontSize:self.style.textSize];
+    
+    // Find the longest line in current closed caption
+    NSArray* lines = [self.caption.text componentsSeparatedByString:@"\n"];
+    CGSize maxLineSize = CGSizeMake(0, 0);
+    for (NSString* line in lines) {
+        CGSize size = [line sizeWithAttributes:@{NSFontAttributeName: self.textView.font}];
+        if (size.width > maxLineSize.width) {
+            maxLineSize = size;
+            if (maxLineSize.width > maxWidth) {
+                break;
+            }
+        }
+    }
+    
+    CGFloat frameWidth = maxWidth;
+    if (maxWidth > maxLineSize.width) {
+        frameWidth = maxLineSize.width;
+    }
+    
+    // Calculate height of the frame based on the width calculated above
+    NSMutableString* resultText = [[NSMutableString alloc] init];
+    NSInteger lineCount = [lines count];
+    
+    // Need to split some lines if those lines are longer than the maxWidth
+    if (maxLineSize.width >= maxWidth) {
+        NSInteger currentLineNum = 0;
+        for (NSString* line in lines) {
+            currentLineNum++;
+            NSMutableString* temp = [NSMutableString stringWithString:line];
+            
+            // Find the last whitespace before exceeding maxWidth and insert a new line char there to split that line
+            int prevWhitespaceIndex = 0;
+            for (int i = 0; i < [line length]; i++) {
```

Comment:
make i unsigned to remove warnings. e.g. `unsigned int i = 0`

Reply:
I see multiple warnings thrown by the compiler here. Can you try to eliminate them?